sympy/tensor/index_methods.py

```diff
@@ -24,6 +23,7 @@ class IndexConformanceException(Exception):
     pass
 
 
+@deprecated(useinstead='EinsteinSum', issue=9284)
 def _remove_repeated(inds):
```

Comment:
> @toolforger : if the function can't easily be doctested, then the
> docs shouldn't be tested.

It's still easy to test. Just add the DeprecationWarning to the output 
being tested.

 > Doctests are for ensuring the docs remain
> valid, not for testing actual functionality.

If the doctests test actual functionality, they should be moved to 
`test_tensor.py`, not skipped.
Rationale is that we need to keep even deprecated functionality in a 
working state, and testing is the tool we use to ensure that, so 
disabling testing a feature is the last-ditch

 > Skipping hard to test docs is perfectly fine.

If the trade-off says it's too hard to be worth it, I agree.
In this case, it isn't even hard to test.

 > Either way, @moorepants  is right here - this
> function is non-public and doesn't need any deprecation warning.

`_remove_repeated` is called from `get_indices`, which is public but 
deprecated.
So `_remove_repeated` cannot be removed without breaking the public API.